operator/src/main/java/oracle/kubernetes/weblogic/domain/model/ServerEnvVars.java

```diff
@@ -62,10 +62,12 @@
   /** If present, pod scripts will watch for changes to override configurations and move them into place. */
   public static final String DYNAMIC_CONFIG_OVERRIDE = "DYNAMIC_CONFIG_OVERRIDE";
 
+  public static final String CLOUD_PLATFORM = "CLOUD_PLATFORM";
+
   private static final List<String> RESERVED_NAMES = Arrays.asList(
         DOMAIN_UID, DOMAIN_NAME, DOMAIN_HOME, NODEMGR_HOME, SERVER_NAME, SERVICE_NAME,
         ADMIN_NAME, AS_SERVICE_NAME, ADMIN_PORT, ADMIN_PORT_SECURE, ADMIN_SERVER_PORT_SECURE,
-        LOG_HOME, SERVER_OUT_IN_POD_LOG, DATA_HOME, ACCESS_LOG_IN_LOG_HOME, DYNAMIC_CONFIG_OVERRIDE);
+        LOG_HOME, SERVER_OUT_IN_POD_LOG, DATA_HOME, ACCESS_LOG_IN_LOG_HOME, DYNAMIC_CONFIG_OVERRIDE, CLOUD_PLATFORM);
```

Comment:
Is this going to cause operator upgrade rolling side effect ?

Reply:
+1

Reply:
I believe it will not cause the rolling side effect, I'll run the upgrade to confirm.

Reply:
I have verified that rolling doesn't happen during upgrade when `kubernetesPlatform` helm variable is not present. I add the `KUBERNETES_PLATFORM` environment variable to server pods only when `kubernetesPlatform` is set in the helm values.